packages/ens-controller/src/EnsController.ts

```diff
@@ -63,19 +74,33 @@ export class EnsController extends BaseControllerV2<
   EnsControllerState,
   EnsControllerMessenger
 > {
+  private _ens?: Ens;
+
   /**
    * Creates an EnsController instance.
    *
    * @param options - Constructor options.
    * @param options.messenger - A reference to the messaging system.
    * @param options.state - Initial state to set on this controller.
+   * @param options.ens - Ens instance.
+   * @param options.provider - Provider config.
+   * @param options.onNetworkDidChange - onNetworkDidChange.
+   * @param options.getCurrentChainId - getCurrentChainId .
    */
   constructor({
     messenger,
-    state,
+    state = {},
+    ens,
+    provider,
+    onNetworkDidChange,
+    getCurrentChainId,
   }: {
     messenger: EnsControllerMessenger;
     state?: Partial<EnsControllerState>;
+    ens?: Ens;
```

Comment:
Taking a look at extension, it doesn't seem that it passes the `ens` property: https://github.com/MetaMask/metamask-extension/blob/92556867f725d8b47b47b9f8f57337531dba6a8f/app/scripts/metamask-controller.js#L606-L614. So is this necessary?

Also, it seems like it always passes `provider`. Should we make that non-optional? (It also passes `onNetworkDidChange` and `getCurrentChainId` too, but I'm ignoring those — see below.)